test/test_dot_device.cc

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) 2017-2022, University of Tennessee. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+// This program is free software: you can redistribute it and/or modify it under
+// the terms of the BSD 3-Clause license. See the accompanying LICENSE file.
+
+#include "test.hh"
+#include "cblas_wrappers.hh"
+#include "lapack_wrappers.hh"
+#include "blas/flops.hh"
+#include "print_matrix.hh"
+
+// -----------------------------------------------------------------------------
+template< typename Tx, typename Ty >
```

Comment:
Follow spacing and capitalization that's in test_dot.cc:
`template <typename TX, typename TY>`

Do `meld test_dot.cc test_dot_device.cc` (with your favorite diff program) and minimizes the differences when comparing files. I pointed out a couple instances below, but a lot of lines should be updated. The only differences should be GPU related.